backend/contracts/ElectionFactory.sol

```diff
@@ -1,28 +1,51 @@
-//SPDX-License-Identifier: UNLICENSED
+// SPDX-License-Identifier: UNLICENSED
 
 pragma solidity ^0.8.0;
 
 contract ElectionFactory {
-    struct Election {
+    struct Candidate {
+        uint8 id;
         string name;
     }
 
-    Election[] public elections;
-
-    constructor() {
+    struct Ballot {
+        string name;
+        string description;
+        uint256 createTime;
+        uint256 votingTime;
+        Candidate[] candidates;
     }
 
-    function createElection(string memory _name) external{
-        Election memory newElection = Election(_name);
-        elections.push(newElection);
+    mapping(uint256 => Ballot) public ballots;
+    uint256 public nextBallotId;
+
+    event CandidateAdded(uint256 ballotId, uint8 id, string name);
+    event ContractInitialized(uint256 ballotId, string name, string description, uint256 createTime, uint256 votingTime, Candidate[] candidates);
+
+    function clearCandidates(uint256 _ballotId) internal {
+        delete ballots[_ballotId].candidates; // just a precaution for any random data that might have left in array
     }
```

Comment:
Is this actually a concern? If so seems like all the people talking that solidity is like JS are really deeply mistaken 😄

Reply:
I actually added this for the first version of the contract because when creating a new vote, the candidate array was populated with the new values instead of being replaced. Needs to check further for this version 🤷‍♀️

Reply:
Double checked, seems like there's no need in this function and everything works properly without it. I'll remove it